rust/lancedb/src/remote/db.rs

```diff
@@ -411,6 +412,12 @@ impl<S: HttpSend> Database for RemoteDatabase<S> {
     }
 }
 
+impl<S: HttpSend> RemoteDatabase<S> {
+    async fn send(&self, req: RequestBuilder) -> Result<(String, Response)> {
+        RemoteTable::send_with_retry(&self.client, req, None, None, true).await
+    }
+}
+
```

Comment:
> This moves retry logic from the HTTP client to the remote table implementation. The reasoning for this is that retries require Arrow serialization which should not be a responsibility of the HTTP client. This aims to keep the REST client concerned with reqwest/HTTP dependencies only.

I'm a little skeptical of this, and it becomes even more awkward now that the database client has to call into the table client API to make an HTTP request.

Maybe keep the `send*` methods in the HTTP client, but can make `send_stream_with_retry` take a closure `FnMut() -> StreamingBody` or something?

Reply:
Yeah, I don't love it, but it definitely doesn't feel right to add RecordBatch stuff to the HTTP client. I also considered the closure idea but thought it might be too complicated. I can give it a try.

Reply:
Done